aphrodite/endpoints/api_server_kobold.py

```diff
@@ -0,0 +1,198 @@
+    input_tokens = tokenizer(kai_payload.prompt).input_ids[-max_input_tokens:]
+
+    return sampling_params, input_tokens
+
+@kai_api.post("/generate")
+async def generate(kai_payload: KAIGenerationInputSchema, raw_request: Request) -> JSONResponse:
+    """ Generate text """
+
+    req_id = f"kai-{random_uuid()}"
+    sampling_params, input_tokens = prepare_engine_payload(kai_payload)
+    result_generator = engine.generate(None, sampling_params, req_id, input_tokens)
+
+    final_res: RequestOutput = None
+    async for res in result_generator:
+        if await raw_request.is_disconnected():
+            # Abort the request if the client disconnects.
+            await engine.abort(req_id)
+            return create_error_response(HTTPStatus.BAD_REQUEST,
+                                         "Client disconnected")
+        final_res = res
+    assert final_res is not None
+
+    return JSONResponse({"results": [{"text": output.text} for output in final_res.outputs]})
+
+
+@extra_api.post("/generate/stream")
+async def generate_stream(kai_payload: KAIGenerationInputSchema) -> StreamingResponse:
+    """ Generate text SSE streaming """
+
+    req_id = f"kai-{random_uuid()}"
+    sampling_params, input_tokens = prepare_engine_payload(kai_payload)
+    results_generator = engine.generate(None, sampling_params, req_id, input_tokens)
+
+    async def stream_kobold() -> AsyncGenerator[bytes, None]:
+        previous_output = ""
+        async for res in results_generator:
+            new_chunk = res.outputs[0].text[len(previous_output):]
+            previous_output += new_chunk
+            yield b"event: message\n"
+            yield f"data: {json.dumps({'token': new_chunk})}\n\n".encode()
+
+    return StreamingResponse(stream_kobold(),
+                             headers={"Cache-Control": "no-cache", "Connection": "keep-alive"},
+                             media_type='text/event-stream')
+
+@kai_api.get("/info/version")
+async def get_version():
+    """ Impersonate KAI """
+    return JSONResponse({"result": "1.2.4"})
+
+@kai_api.get("/model")
+async def get_model():
+    """ Get current model """
+    return JSONResponse({"result": f"aphrodite/{served_model}"})
+
+@kai_api.get("/config/soft_prompts_list")
+async def get_available_softprompts():
+    """ stub for compatibility """
+    return JSONResponse({"values":[]})
+
+@kai_api.get("/config/soft_prompt")
+async def get_current_softprompt():
+    """ stub for compatibility """
+    return JSONResponse({"value": ""})
+
+@kai_api.put("/config/soft_prompt")
+async def set_current_softprompt():
+    """ stub for compatibility """
+    return JSONResponse({})
+
+@extra_api.post("/abort")
+async def abort_generation():
+    """ stub for compatibility """
+    return JSONResponse({})
+
+@extra_api.get("/version")
+async def get_extra_version():
+    """ Impersonate KoboldCpp with streaming support """
+    return JSONResponse({"result": "KoboldCpp", "version": "1.30"})
+
```

Comment:
For maximum compatibility with horde, we may want a route for `/api/latest/config/max_context_length`. We can pull this from `engine_args.max_model_len`.